Microsoft.Azure.Devices.LongHaul.Module/IotHub.cs

```diff
@@ -0,0 +1,490 @@
+        private static readonly TimeSpan s_deviceTwinUpdateInterval = TimeSpan.FromSeconds(3);
+        private static readonly TimeSpan s_retryInterval = TimeSpan.FromSeconds(1);
+        private readonly ConcurrentQueue<TelemetryMessage> _messagesToSend = new();
+
+        private long _totalTelemetryMessagesToModuleSent = 0;
+        private long _totalTwinUpdatesToModuleReported = 0;
+        private long _totalTwinCallbacksToModuleHandled = 0;
+        private long _totalDesiredPropertiesToModuleHandled = 0;
+        private long _totalM2mMessagesCompleted = 0;
+        private long _totalM2mMessagesRejected = 0;
+        private long _totalMethodCallsToModuleCount = 0;
+
+        public IotHub(Logger logger, Parameters parameters)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _moduleConnectionString = parameters.ConnectionString;
+            _clientOptions = new IotHubClientOptions(parameters.GetTransportSettings())
+            {
+                PayloadConvention = parameters.GetPayloadConvention(),
+                SdkAssignsMessageId = SdkAssignsMessageId.WhenUnset,
+            };
+            _moduleClient = null;
+        }
+
+        public bool IsConnected => _moduleClient.ConnectionStatusInfo.Status == ConnectionStatus.Connected;
+
+        public Dictionary<string, string> TelemetryUserProperties { get; } = new();
+
+        /// <summary>
+        /// Initializes the connection to IoT Hub.
+        /// </summary>
+        public async Task InitializeAsync()
+        {
+            await _lifetimeControl.WaitAsync().ConfigureAwait(false);
+
+            try
+            {
+                if (_moduleClient == null)
+                {
+                    _moduleClient = new IotHubModuleClient(_moduleConnectionString, _clientOptions)
+                    {
+                        ConnectionStatusChangeCallback = ConnectionStatusChangesHandlerAsync
+                    };
+                }
+                else
+                {
+                    await _moduleClient.CloseAsync().ConfigureAwait(false);
+                }
+
+                await _moduleClient.OpenAsync().ConfigureAwait(false);
+                await _moduleClient.SetDirectMethodCallbackAsync(DirectMethodCallback).ConfigureAwait(false);
+                await _moduleClient.SetDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallbackAsync).ConfigureAwait(false);
+                await _moduleClient.SetIncomingMessageCallbackAsync(OnM2mMessageReceivedAsync).ConfigureAwait(false);
+            }
+            finally
+            {
+                _lifetimeControl.Release();
+            }
+        }
+
+        public async Task InvokeDirectMethodAsync(Logger logger, CancellationToken ct)
+        {
+            var helper = new IotHubConnectionStringHelper(_moduleConnectionString);
+            logger.LoggerContext.Add(OperationName, DirectMethod);
+            Stopwatch sw = new();
```

Comment:
We typically only use this notation for class fields. Otherwise:

```suggestion
            var sw = new Stopwatch();
```